python/lsst/ip/diffim/utils.py

```diff
@@ -711,3 +732,354 @@ def plotWhisker(results, newWcs):
+        # detectConfig.nSigmaToGrow = psfSigma
+        detectConfig.reEstimateBackground = True  # if False, will fail often for faint sources on gradients?
+        detectConfig.thresholdType = "pixel_stdev"
+
+        # Create the detection task. We pass the schema so the task can declare a few flag fields
+        detectTask = measAlg.SourceDetectionTask(schema, config=detectConfig)
+
+        table = afwTable.SourceTable.make(schema)
+        catalog = detectTask.makeSourceCatalog(table, diffim, sigma=psfSigma)
+
+        # Now do the merge.
+        if doMerge:
+            fpSet = catalog.fpSets.positive
+            fpSet.merge(catalog.fpSets.negative, grow, grow, False)
+            sources = afwTable.SourceCatalog(table)
+            fpSet.makeSources(sources)
+
+            return sources
+
+        else:
+            return detectTask, schema
+
+
+# ######## UTILITIES FUNCTIONS FOR PLOTTING  ####
+
+# Utility class containing methods for displaying dipoles/footprints
+# in difference images, mostly used for debugging.
+
+
+def display2dArray(arr, title='Data', showBars=True, extent=None):
+    """Use matplotlib.pyplot.imshow() to display a 2-D array.
+
+    @param arr The 2-D array to display
+    @param title Optional title to display
+    @param showBars Show grey-scale bar alongside
+    @param extent If not None, a 4-tuple giving the bounding box coordinates of the array
+
+    @return matplotlib.pyplot.figure dispaying the image
+    """
+    plt = importMatplotlib()
+    if not plt:
+        return
+
+    fig = plt.imshow(arr, origin='lower', interpolation='none', cmap='gray',
+                     extent=extent)
+    plt.title(title)
+    if showBars:
+        plt.colorbar(fig, cmap='gray')
+    return fig
+
+
+def displayImage(image, showBars=True, width=8, height=2.5):
+    """Use matplotlib.pyplot.imshow() to display an afw.image.Image within
+    its bounding box.
+
+    @see displayImages()
+    """
+    return displayImages((image), showBars=True, width=8, height=2.5)
+
+
+def displayImages(images, showBars=True, width=8, height=2.5):
+    """Use matplotlib.pyplot.imshow() to display up to three
+    afw.image.Images alongside each other, each within its
+    bounding box.
+
+    @see display2dArray() for params not listed here.
+    @param images tuple of up to three images to display
+    @param image The image to display
+    @param width The width of the display (inches)
+    @param height The height of the display (inches)
+
+    @return matplotlib.pyplot.figure dispaying the image
+
+    """
+    plt = importMatplotlib()
+    if not plt:
+        return
+
+    fig = plt.figure(figsize=(width, height))
+    for i, image in enumerate(images):
+        bbox = image.getBBox()
+        extent = (bbox.getBeginX(), bbox.getEndX(), bbox.getBeginY(), bbox.getEndY())
+        plt.subplot(1, len(images), i+1)
+        ma = image.getArray()
+        display2dArray(ma, title='Data', showBars=showBars, extent=extent)
```

Comment:
Did you try this with an image for which xy0 is not (0,0)? I would expect it to fail because image.getArray() returns a normal numpy array that knows nothing about xy0. I'm really puzzled by what extent is supposed to do. Unless you want to support subregions I suspect it should be ditched here and in display2dArray.